instrumentation/p6spy/src/main/java/brave/p6spy/TracingJdbcEventListener.java

```diff
@@ -66,10 +71,24 @@
     span.start();
   }
 
+  @Override public void onAfterExecuteUpdate(
+    PreparedStatementInformation statementInformation, long timeElapsedNanos, int rowCount, SQLException e
+  ) {
+    Span span = ThreadLocalSpan.CURRENT_TRACER.remove();
+    if (span == null || span.isNoop()) return;
+    if (includeAffectedRowsCount) {
+      span.tag("sql.affected_rows", String.valueOf(rowCount));
+    }
+    finishSpan(span, e);
```

Comment:
Is this right? Do we need to finish Span given that this method did not exist before?

Reply:
The span is started in `onBeforeAnyExecute` for both updates and other SQLs, but then we override `onAfterExecuteUpdate` for updates (instead of them being delegated to `onAfterAnyExecute` from `SimpleJdbcEventListener`). Therefore, we need to finish the span both in `onAfterExecuteUpdate` and `onAfterAnyExecute`.